enterprise/suppress-user/handler.go

```diff
@@ -0,0 +1,51 @@
+package suppression
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/rudderlabs/rudder-server/utils/logger"
+	"github.com/samber/lo"
+)
+
+// newHandler creates a new handler for the suppression feature
+func newHandler(ctx context.Context, r Repository, log logger.Logger) *handler {
```

Comment:
[optional] You don't really need a repository here, more like a `suppresser`:

```go
type suppresser interface {
	// Suppress returns true if the given user is suppressed, false otherwise
	Suppress(workspaceID, userID, sourceID string) (bool, error)
}
```

Reply:
yes, but I see no harm using the repository itself either

Reply:
I noticed because I wanted to mock it to write a test for the handler and I had to provide all the other methods too :man_shrugging:

Reply:
Using the memory repository as a basis can simplify this task

```
type mockRepo struct {
  Repository
}

func (*mockRepo) Suppress(workspaceID, userID, sourceID string) (bool, error) {
   return false, errors.New("zonk!")
}

func newMockRepo() Repository {
 return &mockRepo{NewMemoryRepository(logger.NOP)}
}
```